library/spdm_crypt_lib/libspdm_crypt_asym.c

```diff
@@ -526,6 +526,144 @@ static bool libspdm_asym_func_need_hash(uint32_t base_asym_algo)
     return false;
 }
 
+
+static void libspdm_copy_signature_swap_endian_rsa(
+    uint8_t* dst,
+    size_t dst_size,
+    const uint8_t* src,
+    size_t src_size)
+{
+    /* RSA signature is a single buffer to be swapped */
+    size_t i;
+
+    if (src == dst) {
+        LIBSPDM_ASSERT(dst_size == src_size);
+
+        /* src and dst are same buffer. Swap in place. */
+
+        uint8_t byte;
+        for (i = 0; i < dst_size / 2; i++) {
+            byte = dst[i];
+            dst[i] = dst[dst_size - i - 1];
+            dst[dst_size - i - 1] = byte;
+        }
+    }
+    else {
+        /* src and dst are different non-overlapping buffers */
```

Comment:
`!(src == dst)` doesn't mean that they are non-overlapping.

Reply:
can we just swap in place?

Reply:
@steven-bellock  Right. Perhaps I should have said this function only supports in-place same buffer swap, or completely different buffers (no overlap). That is what the asserts were supposed to do on line 544-555.

@jyao1 I can do in place, but I need to cast away const-ness, or change the parameter type in a higher up function to remove the const.  Plus it leaves the original buffer modified/swapped unless I swap it back. Feels like a bad side-effect to me, but I can see the benefit of saving the additional stack space, which can be considerable for some cases (512 bytes RSA 4072). I'm open to whatever you prefer.

Reply:
@jyao1 Waiting for your comments on this.

Reply:
I think it is OK to have it, to support additional buffer.

Reply:
@steven-bellock Updated the comments in my latest push to try to be clear. Overlap is not a supported usage model, except when they are the exact same buffer.